internal/provider/resource_aci_pim_route_map_entry.go

```diff
@@ -0,0 +1,597 @@
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	tflog.Trace(ctx, fmt.Sprintf("Delete of resource aci_pim_route_map_entry with id '%s'", data.Id.ValueString()))
+	jsonPayload, message, messageDetail := getPimRouteMapEntryDeleteJsonPayload(ctx, data)
+	if jsonPayload == nil {
+		resp.Diagnostics.AddError(message, messageDetail)
+		return
+	}
+	requestData, message, messageDetail := doPimRouteMapEntryRequest(ctx, r.client, fmt.Sprintf("api/mo/%s.json", data.Id.ValueString()), "POST", jsonPayload)
+	if requestData == nil {
+		resp.Diagnostics.AddError(message, messageDetail)
+		return
+	}
+	tflog.Trace(ctx, "End delete of resource: aci_pim_route_map_entry")
+}
+
+func (r *PimRouteMapEntryResource) ImportState(ctx context.Context, req resource.ImportStateRequest, resp *resource.ImportStateResponse) {
+	resource.ImportStatePassthroughID(ctx, path.Root("id"), req, resp)
+}
+
+func setPimRouteMapEntryAttributes(ctx context.Context, client *client.Client, data *PimRouteMapEntryResourceModel) interface{} {
+	requestData, message, messageDetail := doPimRouteMapEntryRequest(ctx, client, fmt.Sprintf("api/mo/%s.json?rsp-subtree=children&rsp-subtree-class=%s", data.Id.ValueString(), "pimRouteMapEntry,tagAnnotation"), "GET", nil)
+
+	if requestData == nil {
+		return map[string]string{"message": message, "messageDetail": messageDetail}
+	}
+	if requestData.Search("imdata").Search("pimRouteMapEntry").Data() != nil {
+		classReadInfo := requestData.Search("imdata").Search("pimRouteMapEntry").Data().([]interface{})
+		if len(classReadInfo) == 1 {
+			attributes := classReadInfo[0].(map[string]interface{})["attributes"].(map[string]interface{})
+			for attributeName, attributeValue := range attributes {
+				if attributeName == "dn" {
+					data.Id = basetypes.NewStringValue(attributeValue.(string))
+					setPimRouteMapEntryParentDn(ctx, attributeValue.(string), data)
+				}
+				if attributeName == "action" {
+					data.Action = basetypes.NewStringValue(attributeValue.(string))
+				}
+				if attributeName == "annotation" {
+					data.Annotation = basetypes.NewStringValue(attributeValue.(string))
+				}
+				if attributeName == "descr" {
+					data.Descr = basetypes.NewStringValue(attributeValue.(string))
+				}
+				if attributeName == "grp" {
+					data.Grp = basetypes.NewStringValue(attributeValue.(string))
+				}
+				if attributeName == "name" {
+					data.Name = basetypes.NewStringValue(attributeValue.(string))
+				}
+				if attributeName == "nameAlias" {
+					data.NameAlias = basetypes.NewStringValue(attributeValue.(string))
+				}
+				if attributeName == "order" {
+					data.Order = basetypes.NewStringValue(attributeValue.(string))
+				}
+				if attributeName == "rp" {
+					data.Rp = basetypes.NewStringValue(attributeValue.(string))
+				}
+				if attributeName == "src" {
+					data.Src = basetypes.NewStringValue(attributeValue.(string))
+				}
+			}
+			TagAnnotationPimRouteMapEntryList := make([]TagAnnotationPimRouteMapEntryResourceModel, 0)
```

Comment:
Just a suggestion, we could have extractTagChildren in all the resources:
   tagAnnotationList, err := extractTagChildren(classReadInfo)
    if err != nil {
        return err
    }

Have this in utils->
    func extractTagChildren(classReadInfo map[string]interface{}) {
    
}

Reply:
to be done at later stage